.NET Guide - Web Essentials (#145)
JetBrains/jetbrains_guide: sites/dotnet-guide/contents/tutorials/web-fundamentals/debugging/index.md

---
type: TutorialStep
date: 2020-12-01
title: Debugging code
technologies: [asp.net, javascript]
topics: [web,debugging]
products: [rider]
author: pwe
subtitle: What's your code doing? Step through code with the debugger, and find out. 
thumbnail: ./thumbnail.png
longVideo:
    poster: ./poster.png
    url: https://youtu.be/todo
---

> Sometimes code has bugs, and you need to debug it. For JavaScript code, our IDEs bundle WebStorm’s debugger. Let’s take a look at its key features.

Reviewer: ```suggestion
Even the best codebases have bugs. For JavaScript developers, our IDEs bundle WebStorm’s debugger. Let’s take a look at its key features.
```

Let’s open an HTML page using the IDE. From the context menu in the **Solution** tool window, choose **Debug**.

> In HTML files, this creates and selects a JavaScript Run/Debug configuration, which is a **JavaScript Debug** configuration type. It points to our project’s `index.html` page. The configuration uses Chrome as the browser.

Reviewer: ```suggestion
In HTML files, this creates and selects a JavaScript Run/Debug configuration, which is a **JavaScript Debug** configuration type. The run configuration points to our project’s `index.html` page with Chrome as the browser.
```
